src/views/TradingCompetition/index.tsx

```diff
@@ -115,10 +135,49 @@ const TradingCompetition = () => {
         canClaimNFT: false,
       })
     }
-
-    fetchCompetitionInfo()
+    fetchCompetitionInfoContract()
   }, [account, registrationSuccessful, claimSuccessful, tradingCompetitionContract])
 
+  useEffect(() => {
+    const fetchUserTradingStats = async () => {
+      const res = await fetch(`${profileApiUrl}/api/users/${account}`)
+      const data = await res.json()
+      setUserLeaderboardInformation(data.leaderboard)
+    }
+
+    const fetchGlobalLeaderboardStats = async () => {
+      const res = await fetch(`${profileApiUrl}/api/leaderboard/global`)
+      const data = await res.json()
+      setGlobalLeaderboardInformation(data)
+    }
+
+    const fetchTeamsLeaderboardStats = async (teamId, callBack) => {
+      try {
+        const res = await fetch(`${profileApiUrl}/api/leaderboard/team/${teamId}`)
+        const data = await res.json()
+        callBack(data)
+      } catch (e) {
+        console.error(e)
+      }
+    }
+
+    if (account) {
+      fetchUserTradingStats()
+    }
+
+    fetchTeamsLeaderboardStats(1, (data) => setTeam1LeaderboardInformation(data))
+    // CURRENTLY NO TEAM 2 DATA - SO IT BORKS
+    // fetchTeamsLeaderboardStats(2, (data) => setTeam2LeaderboardInformation(data))
+    fetchTeamsLeaderboardStats(3, (data) => setTeam3LeaderboardInformation(data))
+
+    fetchGlobalLeaderboardStats()
```

Comment:
I recommend using `Promise.all`.

```
const [teamLeaderboardStats1, teamLeaderboardStats2, teamLeaderboardStats3] = await Promise.all([
  fetchTeamsLeaderboardStats(1),
  fetchTeamsLeaderboardStats(2),
  fetchTeamsLeaderboardStats(3),
])
```

Reply:
Perhaps you want to organize these into two separate effects. Or make a comment why you are fetching the leaderboard stats all the time, but only user trading stats when an account is present. Just a thought.

Reply:
Yea both ace suggestions cheers Hachioji

Reply:
Separated the useEffects depending on whether they ought to always be fetched, vs only fetching when a user has registered. The callback functions have also changed slightly. DMed you about improving these post-launch